docs/docs/resources/pipeline.yaml

```diff
@@ -0,0 +1,334 @@
+  # add the key-value pairs they need. The fields that are specific to kafka
+  # apps are marked with `# kafka-app-specific`
+  app: # required
+    streams: # required, kafka-app-specific
+      brokers: ${broker} # required
+      schemaRegistryUrl: ${schema_registry_url}
+    nameOverride: override-with-this-name # kafka-app-specific
+    key1:
+      key4: value3
+    key2: value1
+    key3: value2
+  # Topic(s) from which the component will read input
+  from:
+    topics: # required
+      ${pipeline_name}-input-topic:
+        type: input # required
+        role: topic-role
+  # Topic(s) into which the component will write output
+  to:
+    topics: # required
+      ${pipeline_name}-output-topic:
+        type: output # required
+        role: topic-role
+        keySchema: key-schema
+        valueSchema: value-schema
+        partitions_count: 1
+        replication_factor: 1
+        configs: 
+          config1: config1
+    models: 
+      model: model
+  # Pipeline prefix that will prefix every component name. If you wish to not 
+  # have any prefix you can specify an empty string.
+  prefix: ${pipeline_name}-
+  # Helm repository configuration, the default value is given here
+  repoConfig:
+    repositoryName: bakdata-streams-bootstrap # required
+    url: https://bakdata.github.io/streams-bootstrap/ # required
+    repo_auth_flags:
+      username: user
+      password: pass
+      ca_file: /home/user/path/to/ca-file
+      insecure_skip_tls_verify: false
+  version: "2.7.0"
+
+# StreamsApp component that configures a streams bootstrap app.
+# More documentation on StreamsApp: https://github.com/bakdata/streams-bootstrap
+- type: streams-app # required
+  name: streams-app # required
+  namespace: namespace # required
+  # `app` can contain application-specific settings, hence  the user is free to
+  # add the key-value pairs they need. The fields that are specific to streams
+  # apps are marked with `# streams-app-specific`
+  app: # required
+    # Streams Bootstrap streams section
+    streams: # required, streams-app-specific
+      brokers: ${broker} # required
+      schemaRegistryUrl: ${schema_registry_url}
+      inputTopics:
+        - "topic1"
+        - "topic2"
+      outputTopic: output-topic
+      inputPattern: input-pattern
+      extraInputTopics:
+        input_role1:
+          - input_topic1
+          - input_topic2
+        input_role2:
+          - input_topic3
+          - input_topic4
+      extraInputPatterns:
+        pattern_role1: input_pattern1
+      extraOutputTopics: 
+        output_role1: output_topic1
+        output_role2: output_topic2
+      errorTopic: error-topic
+      config: 
+        my.streams.config: my.value
+    nameOverride: override-with-this-name # streams-app-specific
+    autoscaling: # streams-app-specific
+      consumergroup: consumer-group # required
+      topics:
+        - "topic1"
+        - "topic2"
+    key1:
```

Comment:
This follows https://github.com/bakdata/streams-bootstrap/tree/master/charts/streams-app and therefore does not have arbitrary keys. Please also refer to the chart in order to make clear, which configs are allowed

Reply:
Included